File: src/main/java/kdh/boardproject/repository/CategoryRepository.java
package kdh.boardproject.repository;

import kdh.boardproject.entity.Category;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

public interface CategoryRepository extends JpaRepository<Category, Long> {
    List<Category> findAll();
    Optional<Category> findOneByCategoryName(String categoryName);

    Optional<Category> findOneByIdx(Long idx);
}

File: src/main/java/kdh/boardproject/repository/CategoryRepository.kt
package kdh.boardproject.repository

import kdh.boardproject.entity.Category
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CategoryRepository : JpaRepository<Category, Long> {
    override fun findAll(): List<Category>
    fun findOneByCategoryName(categoryName: String): Category?
    fun findOneByIdx(idx: Long): Category?
}